### ISSUE TITLE
PSI test fails manually and works automatically

### ISSUE THREAD

**Issue by latot** [cbm755/octsympy#638]
Hi all, well, here a very weird thing, if i run the test it works, all fine, but if run the same commands in the octave terminal it fails....
```
octave:2> test @sym/psi
PASSES 9 out of 9 tests
octave:3> syms x
octave:4> f = psi(x)
f = (sym) polygamma(0, x)
octave:5> function_handle(f)
error: function_handle: python codegen failed: polygamma(0, x)
error: called from
    function_handle at line 166 column 9
```
the full test:
```
%!test
%! % round trip
%! if (exist ('psi','builtin'))
%! syms x
%! f = psi (x);
%! h = function_handle (f);
%! A = h (1.1);
%! B = psi (1.1);
%! assert (A, B)
%! end
```

Thx. Cya.

**Comment by mtmiller** [comment]
Are you testing with Octave 4.0 or 4.2 (or 4.3)?

With Octave 4.0, the tests pass because `psi` does not exist as a built-in function, so there is nothing to test.

With Octave 4.2.0-rc2, I get a failure with the same error message that you show from running the commands manually.

**Comment by latot** [comment]
Hi, sorry o forgot specify that, to reproduce this you need run octave in Python3, and i test this in octave 4.3, with this the test pass when i run the test command, but fails if i run the command manually.

**Comment by mtmiller** [comment]
I get the same error when I test with Octave 4.3 and either Python 2 or 3

    >> test @sym/psi
    ***** test
     % round trip
     if (exist ('psi','builtin'))
     syms x
     f = psi (x);
     h = function_handle (f);
     A = h (1.1);
     B = psi (1.1);
     assert (A, B)
     end
    !!!!! test failed
    function_handle: python codegen failed: polygamma(0, x)

**Comment by latot** [comment]
mm, let correct the initial conditions of this:
Python2
```
octave:2> test @sym/psi
PASSES 10 out of 10 tests
octave:3> python_cmd('return sys.version')
warning: number of outputs don't match, was this intentional?
warning: called from
    python_cmd at line 186 column 5
ans = 2.7.10 (default, Nov 19 2015, 11:51:46) 
[GCC 4.9.3]
```
Python3
```
octave:2> test @sym/psi
***** test
 % round trip
 if (exist ('psi','builtin'))
 syms x
 f = psi (x);
 h = function_handle (f);
 A = h (1.1);
 B = psi (1.1);
 assert (A, B)
 end
!!!!! test failed
function_handle: python codegen failed: polygamma(0, x)
octave:3> python_cmd('return sys.version')
warning: number of outputs don't match, was this intentional?
warning: called from
    python_cmd at line 186 column 5
ans = 3.4.3 (default, Jan  4 2016, 17:13:16) 
[GCC 4.9.3]
```

Thx. Cya.

**Comment by mtmiller** [comment]
Same versions of SymPy in both Python environments? Same Octave 4.3 build in both tests?

**Comment by latot** [comment]
Yes, i have sympy and octave with ebuilds, so system versions, the only change is the PYTHON var to run octave.

**Comment by latot** [comment]
No wait.., let me check better this:
Python 3:
```
octave:4> python_cmd('return spver')
warning: number of outputs don't match, was this intentional?
warning: called from
    python_cmd at line 186 column 5
ans = 1.0
```
Python 2:
```
octave:7> python_cmd('return spver')
warning: number of outputs don't match, was this intentional?
warning: called from
    python_cmd at line 186 column 5
ans = 1.0.1.dev
```
So seems python don't is using the right version???, i don't know this is weird, i only have one version of sympy installed, or different expressions with different python versions?

**Comment by latot** [comment]
Hi, first, 2 problems, seems i had installed a local version of sympy, so python3 was using sympy 1.0, now, this is a sympy bug in the version 1, but is fixed in 1.0.1.
Thx @mtmiller for your help!

Cya.
